[Kura 3.0.0 M1 QA] Wires DB Store partial cleanup

If periodic cleanup is configured with 2 elements remaining, this exception is reported and table is not cleaned up:
```
2017-02-21 12:44:04,077 [pool-22-thread-1] ERROR o.e.k.i.w.s.DbWireRecordStore - Error in truncating the table "WR_data"....
java.sql.SQLSyntaxErrorException: unexpected token: 2
        at org.hsqldb.jdbc.JDBCUtil.sqlException(Unknown Source)
        at org.hsqldb.jdbc.JDBCUtil.sqlException(Unknown Source)
        at org.hsqldb.jdbc.JDBCPreparedStatement.<init>(Unknown Source)
        at org.hsqldb.jdbc.JDBCConnection.prepareStatement(Unknown Source)
        at org.eclipse.kura.internal.wire.common.DbServiceHelper.execute(DbServiceHelper.java:125)
        at org.eclipse.kura.internal.wire.store.DbWireRecordStore.clear(DbWireRecordStore.java:255)
        at org.eclipse.kura.internal.wire.store.DbWireRecordStore.access$0(DbWireRecordStore.java:237)
        at org.eclipse.kura.internal.wire.store.DbWireRecordStore$1.run(DbWireRecordStore.java:225)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
Caused by: org.hsqldb.HsqlException: unexpected token: 2
        at org.hsqldb.error.Error.parseError(Unknown Source)
        at org.hsqldb.ParserBase.unexpectedToken(Unknown Source)
        at org.hsqldb.ParserCommand.compileStatement(Unknown Source)
        at org.hsqldb.Session.compileStatement(Unknown Source)
        at org.hsqldb.StatementManager.compile(Unknown Source)
        at org.hsqldb.Session.execute(Unknown Source)
        ... 13 more
```

The reason is that LIMIT is only supported in HSQL since v.2.3.3, according to [1], and Kura 3.0.0 M1 uses v.2.3.0.

The appropriate equivalent SQL for this version would be `DELETE FROM test WHERE rownum() <= 2`.

But it still doesn't solve the fundamental flaw in this SQL statement - LIMIT is used to limit the number of deleted rows, not limit the number of kept rows.

If you want to keep the configured number of rows (i.e. 2), the v.2.3.0 SQL would be this:
`DELETE FROM test WHERE rownum() <= (SELECT count(*) FROM test) - 2`

[1] http://www.hsqldb.org/doc/2.0/changelist_2_0.txt

Kura 3.0.0-SNAPSHOT, build 945

--- a/kura/org.eclipse.kura.wire.component.provider/src/main/java/org/eclipse/kura/internal/wire/store/DbWireRecordStore.java
+++ b/kura/org.eclipse.kura.wire.component.provider/src/main/java/org/eclipse/kura/internal/wire/store/DbWireRecordStore.java
@@ -83,7 +83,7 @@ public final class DbWireRecordStore implements WireEmitter, WireReceiver, Confi
 
     private static final String SQL_CREATE_TABLE = "CREATE TABLE IF NOT EXISTS {0} (TIMESTAMP BIGINT NOT NULL PRIMARY KEY);";
 
-    private static final String SQL_DELETE_RANGE_TABLE = "DELETE FROM {0} LIMIT {1};";
+    private static final String SQL_DELETE_RANGE_TABLE = "DELETE FROM {0} WHERE rownum() <= (SELECT count(*) FROM {0}) - {1};";
 
     private static final String SQL_DROP_COLUMN = "ALTER TABLE {0} DROP COLUMN {1};";
 

